### ISSUE TITLE
Question About Compiling Without/Removing Classic DOOM

### ISSUE THREAD

**Issue by deuxsonic** [RobertBeckebans/RBDOOM-3-BFG#248]
![removal](https://cloud.githubusercontent.com/assets/7662909/10559636/b0dc0ecc-74c3-11e5-9374-c635d82d1e0f.jpg)

I'm wondering if there's a way to compile this leaving out the classic DOOM stuff as I don't use it as GZDOOM does a better job I think and removing the menu references for changing the game. Can anyone help me do this to my copy of the source? I'm not much of a programmer but I thinker a bit, so I'm sure you guys would know how to remove the game select screen and switch game button on the exit prompt and leaving doomclassic out of the compile if it doesn't break anything for a smaller binary.

**Comment by deuxsonic** [comment]
Also, when I try to compile (win32, VC++ 2010 express) I get this error:

rbdoom-3-bfg-master\neo\libs\ffmpeg-win32\include\libavutil\common.h(34): fatal error C1083: Cannot open include file: 'inttypes.h': No such file or directory

**Comment by deuxsonic** [comment]
1>------ Rebuild All started: Project: ZERO_CHECK, Configuration: Release Win32 ------
1>  Checking Build System
1>  CMake does not need to re-run because C:/RBDOOM-3-BFG-master/build/CMakeFiles/generate.stamp is up-to-date.
1>  CMake does not need to re-run because C:/RBDOOM-3-BFG-master/build/idlib/CMakeFiles/generate.stamp is up-to-date.
2>------ Rebuild All started: Project: idlib, Configuration: Release Win32 ------
2>  Building Custom Rule C:/RBDOOM-3-BFG-master/neo/idlib/CMakeLists.txt
2>  CMake does not need to re-run because C:\RBDOOM-3-BFG-master\build\idlib\CMakeFiles\generate.stamp is up-to-date.
2>  precompiled.cpp
2>  Base64.cpp
2>  BitMsg.cpp
2>  CmdArgs.cpp
2>  CommandLink.cpp
2>  Dict.cpp
2>  Heap.cpp
2>  LangDict.cpp
2>  Lexer.cpp
2>  Lib.cpp
2>  MapFile.cpp
2>  ParallelJobList.cpp
2>  Parser.cpp
2>  RectAllocator.cpp
2>  Str.cpp
2>  Thread.cpp
2>  Timer.cpp
2>  Token.cpp
2>  Bounds.cpp
2>  Box.cpp
2>  Sphere.cpp
2>  HashIndex.cpp
2>  DrawVert.cpp
2>  JointTransform.cpp
2>  Surface.cpp
2>  Surface_Patch.cpp
2>  Surface_Polytope.cpp
2>  Surface_SweptSpline.cpp
2>  TraceModel.cpp
2>  Winding.cpp
2>  Winding2D.cpp
2>  CRC32.cpp
2>  MD4.cpp
2>  MD5.cpp
2>  Angles.cpp
2>  Complex.cpp
2>  Lcp.cpp
2>  Math.cpp
2>  Matrix.cpp
2>  MatX.cpp
2>  Ode.cpp
2>  Plane.cpp
2>  Pluecker.cpp
2>  Polynomial.cpp
2>  Quat.cpp
2>  Rotation.cpp
2>  Simd.cpp
2>  Simd_Generic.cpp
2>..\..\neo\idlib\math\Simd_Generic.cpp(47): warning C4005: 'NODEFAULT' : macro redefinition
2>          c:\rbdoom-3-bfg-master\neo\idlib\sys/sys_types.h(168) : see previous definition of 'NODEFAULT'
2>  Simd_SSE.cpp
2>  Vector.cpp
2>  VecX.cpp
2>  sys_assert.cpp
2>  win_thread.cpp
2>  SoftwareCache.cpp
2>  RenderMatrix.cpp
2>  idlib.vcxproj -> C:\RBDOOM-3-BFG-master\build\idlib\Release\idlib.lib
3>------ Rebuild All started: Project: RBDoom3BFG, Configuration: Release Win32 ------
3>  Building Custom Rule C:/RBDOOM-3-BFG-master/neo/CMakeLists.txt
3>  CMake does not need to re-run because C:\RBDOOM-3-BFG-master\build\CMakeFiles\generate.stamp is up-to-date.
3>  precompiled.cpp
3>  AASBuild.cpp
3>  AASBuild_file.cpp
3>  AASBuild_gravity.cpp
3>  AASBuild_ledge.cpp
3>  AASBuild_merge.cpp
3>  AASCluster.cpp
3>  AASReach.cpp
3>  Brush.cpp
3>  BrushBSP.cpp
3>  dmap.cpp
3>  facebsp.cpp
3>  gldraw.cpp
3>  glfile.cpp
3>  leakfile.cpp
3>  map.cpp
3>  optimize.cpp
3>  portals.cpp
3>  tritjunction.cpp
3>  tritools.cpp
3>  ubrush.cpp
3>  usurface.cpp
3>  AASFile.cpp
3>  AASFileManager.cpp
3>  AASFile_optimize.cpp
3>  AASFile_sample.cpp
3>  CollisionModel_contacts.cpp
3>  CollisionModel_contents.cpp
3>  CollisionModel_debug.cpp
3>  CollisionModel_files.cpp
3>  CollisionModel_load.cpp
3>  CollisionModel_rotate.cpp
3>  CollisionModel_trace.cpp
3>  CollisionModel_translate.cpp
3>  CmdSystem.cpp
3>  Common_demos.cpp
3>  Common_dialog.cpp
3>  common_frame.cpp
3>  Common_load.cpp
3>  Common_localize.cpp
3>  Common_menu.cpp
3>  Common_network.cpp
3>  Common_printf.cpp
3>  Compressor.cpp
3>  Console.cpp
3>  ConsoleHistory.cpp
3>  CVarSystem.cpp
3>  DebugGraph.cpp
3>  DeclAF.cpp
3>  DeclEntityDef.cpp
3>  DeclFX.cpp
3>  DeclManager.cpp
3>  DeclParticle.cpp
3>  DeclPDA.cpp
3>  DeclSkin.cpp
3>  DeclTable.cpp
3>  DemoFile.cpp
3>  EditField.cpp
3>  EventLoop.cpp
3>  File.cpp
3>  FileSystem.cpp
3>  File_Manifest.cpp
3>  File_Resource.cpp
3>  File_SaveGame.cpp
3>  KeyInput.cpp
3>  PlayerProfile.cpp
3>  TokenParser.cpp
3>  UsercmdGen.cpp
3>  BinaryImage.cpp
3>  BoundsTrack.cpp
3>  BufferObject.cpp
3>  Cinematic.cpp
3>  Font.cpp
3>c:\rbdoom-3-bfg-master\neo\libs\ffmpeg-win32\include\libavutil\common.h(34): fatal error C1083: Cannot open include file: 'inttypes.h': No such file or directory
3>  Framebuffer.cpp
3>  GLMatrix.cpp
3>  GuiModel.cpp
3>  ImageManager.cpp
3>  Image_files.cpp
3>  Image_intrinsic.cpp
3>  Image_load.cpp
3>  Image_process.cpp
3>  Image_program.cpp
3>  Interaction.cpp
3>  Material.cpp
3>  Model.cpp
3>  ModelDecal.cpp
3>  ModelManager.cpp
3>  ModelOverlay.cpp
3>  Model_ase.cpp
3>  Model_beam.cpp
3>  Model_liquid.cpp
3>  Model_lwo.cpp
3>  Model_ma.cpp
3>  Model_md3.cpp
3>  Model_md5.cpp
3>  Model_prt.cpp
3>  Model_sprite.cpp
3>  RenderEntity.cpp
3>  RenderProgs.cpp
3>  RenderProgs_GLSL.cpp
3>  RenderSystem.cpp
3>..\neo\renderer\RenderProgs_GLSL.cpp(1160): warning C4065: switch statement contains 'default' but no 'case' labels
3>  RenderSystem_init.cpp
3>  RenderWorld.cpp
3>..\neo\renderer\RenderSystem_init.cpp(1898): warning C4101: 'outSize' : unreferenced local variable
3>  RenderWorld_defs.cpp
3>  RenderWorld_demo.cpp
3>  RenderWorld_load.cpp
3>  RenderWorld_portals.cpp
3>  ResolutionScale.cpp
3>  ScreenRect.cpp
3>  tr_backend_draw.cpp
3>  tr_backend_rendertools.cpp
3>  tr_frontend_addlights.cpp
3>  tr_frontend_addmodels.cpp
3>  tr_frontend_deform.cpp
3>  tr_frontend_guisurf.cpp
3>  tr_frontend_main.cpp
3>  tr_frontend_subview.cpp
3>  tr_trace.cpp
3>  tr_trisurf.cpp
3>  VertexCache.cpp
3>  ColorSpace.cpp
3>  gl_backend.cpp
3>  gl_GraphicsAPIWrapper.cpp
3>  gl_Image.cpp
3>  unzip.cpp
3>  snd_emitter.cpp
3>  snd_shader.cpp
3>  snd_system.cpp
3>  snd_world.cpp
3>  SoundVoice.cpp
3>  WaveFile.cpp
3>  BindWindow.cpp
3>  ChoiceWindow.cpp
3>  DeviceContext.cpp
3>  EditWindow.cpp
3>  FieldWindow.cpp
3>  GameBearShootWindow.cpp
3>  GameBustOutWindow.cpp
3>  GameSSDWindow.cpp
3>  GuiScript.cpp
3>  ListGUI.cpp
3>  ListWindow.cpp
3>  RegExp.cpp
3>  RenderWindow.cpp
3>  SimpleWindow.cpp
3>  SliderWindow.cpp
3>  UserInterface.cpp
3>  Window.cpp
3>  Winvar.cpp
3>  SWF_Bitstream.cpp
3>  SWF_Dictionary.cpp
3>  SWF_Events.cpp
3>  SWF_Image.cpp
3>  SWF_Load.cpp
3>  SWF_Main.cpp
3>  SWF_Names.cpp
3>  SWF_ParmList.cpp
3>  SWF_PlaceObject.cpp
3>  SWF_Render.cpp
3>  SWF_ScriptFunction.cpp
3>  SWF_ScriptObject.cpp
3>  SWF_ScriptVar.cpp
3>  SWF_ShapeParser.cpp
3>  SWF_Shapes.cpp
3>  SWF_Sounds.cpp
3>  SWF_SpriteInstance.cpp
3>  SWF_Sprites.cpp
3>  SWF_Text.cpp
3>  SWF_TextInstance.cpp
3>  SWF_Zlib.cpp
3>  localuser.cpp
3>  session_local.cpp
3>  signin.cpp
3>  socket_net.cpp
3>  am_map.cpp
3>  doomdef.cpp
3>  doominterface.cpp
3>  doomlib.cpp
3>  doomstat.cpp
3>  dstrings.cpp
3>  d_items.cpp
3>  d_main.cpp
3>  d_net.cpp
3>  f_finale.cpp
3>  f_wipe.cpp
3>  globaldata.cpp
3>  g_game.cpp
3>  hu_lib.cpp
3>  hu_stuff.cpp
3>  info.cpp
3>  i_input.cpp
3>  i_main.cpp
3>  i_net.cpp
3>  i_net_win32.cpp
3>  i_system.cpp
3>  i_video_ps3.cpp
3>  mus2midi.cpp
3>  m_argv.cpp
3>  m_bbox.cpp
3>  m_cheat.cpp
3>  m_fixed.cpp
3>  m_menu.cpp
3>  m_misc.cpp
3>  m_random.cpp
3>  m_swap.cpp
3>  p_ceilng.cpp
3>  p_doors.cpp
3>  p_enemy.cpp
3>  p_floor.cpp
3>  p_inter.cpp
3>  p_lights.cpp
3>  p_map.cpp
3>  p_maputl.cpp
3>  p_mobj.cpp
3>  p_plats.cpp
3>  p_pspr.cpp
3>  p_saveg.cpp
3>  p_setup.cpp
3>  p_sight.cpp
3>  p_spec.cpp
3>  p_switch.cpp
3>  p_telept.cpp
3>..\doomclassic\doom\p_pspr.cpp(586): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>  p_tick.cpp
3>  p_user.cpp
3>  r_bsp.cpp
3>  r_data.cpp
3>..\doomclassic\doom\r_bsp.cpp(295): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>..\doomclassic\doom\r_bsp.cpp(435): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>..\doomclassic\doom\p_user.cpp(210): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>  r_draw.cpp
3>  r_main.cpp
3>  r_plane.cpp
3>..\doomclassic\doom\r_main.cpp(315): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>  r_segs.cpp
3>  r_sky.cpp
3>  r_things.cpp
3>  sounds.cpp
3>..\doomclassic\doom\r_segs.cpp(595): warning C4146: unary minus operator applied to unsigned type, result still unsigned
3>  st_lib.cpp
3>  st_stuff.cpp
3>  s_sound.cpp
3>  v_video.cpp
3>  wi_stuff.cpp
3>  w_wad.cpp
3>  z_zone.cpp
3>  Actor.cpp
3>  AF.cpp
3>  AFEntity.cpp
3>  AimAssist.cpp
3>  BrittleFracture.cpp
3>  Camera.cpp
3>  Entity.cpp
3>  Fx.cpp
3>  GameEdit.cpp
3>  Game_local.cpp
3>  Game_network.cpp
3>  Grabber.cpp
3>  IK.cpp
3>  Item.cpp
3>  Leaderboards.cpp
3>  Light.cpp
3>  Misc.cpp
3>  Moveable.cpp
3>  Mover.cpp
3>  MultiplayerGame.cpp
3>  Player.cpp
3>  PlayerIcon.cpp
3>  PlayerView.cpp
3>  Projectile.cpp
3>  Pvs.cpp
3>  SecurityCamera.cpp
3>  SmokeParticles.cpp
3>  Sound.cpp
3>  Target.cpp
3>  Trigger.cpp
3>  Weapon.cpp
3>  WorldSpawn.cpp
3>  AAS.cpp
3>  AAS_debug.cpp
3>  AAS_pathing.cpp
3>  AAS_routing.cpp
3>  AI.cpp
3>  AI_events.cpp
3>  AI_pathing.cpp
3>  AI_Vagary.cpp
3>  Anim.cpp
3>  Anim_Blend.cpp
3>  Anim_Testmodel.cpp
3>  Class.cpp
3>  Event.cpp
3>  SysCmds.cpp
3>  SysCvar.cpp
3>  MenuHandler.cpp
3>  MenuHandler_HUD.cpp
3>  MenuHandler_PDA.cpp
3>  MenuHandler_Scoreboard.cpp
3>  MenuHandler_Shell.cpp
3>  MenuScreen.cpp
3>  MenuScreen_HUD.cpp
3>  MenuScreen_PDA_Inventory.cpp
3>  MenuScreen_PDA_UserData.cpp
3>  MenuScreen_PDA_UserEmails.cpp
3>  MenuScreen_PDA_VideoDisks.cpp
3>  MenuScreen_Scoreboard.cpp
3>  MenuScreen_Shell_Bindings.cpp
3>  MenuScreen_Shell_Browser.cpp
3>  MenuScreen_Shell_ControllerLayout.cpp
3>  MenuScreen_Shell_Controls.cpp
3>  MenuScreen_Shell_Credits.cpp
3>  MenuScreen_Shell_Dev.cpp
3>  MenuScreen_Shell_Difficulty.cpp
3>  MenuScreen_Shell_GameLobby.cpp
3>  MenuScreen_Shell_GameOptions.cpp
3>  MenuScreen_Shell_Gamepad.cpp
3>  MenuScreen_Shell_Leaderboards.cpp
3>  MenuScreen_Shell_Load.cpp
3>  MenuScreen_Shell_MatchSettings.cpp
3>  MenuScreen_Shell_ModeSelect.cpp
3>  MenuScreen_Shell_NewGame.cpp
3>  MenuScreen_Shell_PartyLobby.cpp
3>  MenuScreen_Shell_Pause.cpp
3>  MenuScreen_Shell_Playstation.cpp
3>  MenuScreen_Shell_PressStart.cpp
3>  MenuScreen_Shell_Resolution.cpp
3>  MenuScreen_Shell_Root.cpp
3>  MenuScreen_Shell_Save.cpp
3>  MenuScreen_Shell_Settings.cpp
3>  MenuScreen_Shell_Singleplayer.cpp
3>  MenuScreen_Shell_Stereoscopics.cpp
3>  MenuScreen_Shell_SystemOptions.cpp
3>  MenuWidget.cpp
3>  MenuWidget_Button.cpp
3>  MenuWidget_Carousel.cpp
3>  MenuWidget_CommandBar.cpp
3>  MenuWidget_DynamicList.cpp
3>  MenuWidget_Help.cpp
3>  MenuWidget_InfoBox.cpp
3>  MenuWidget_ItemAssignment.cpp
3>  MenuWidget_List.cpp
3>  MenuWidget_LobbyList.cpp
3>  MenuWidget_MenuBar.cpp
3>  MenuWidget_NavBar.cpp
3>  MenuWidget_NavButton.cpp
3>  MenuWidget_PDA_AudioFiles.cpp
3>  MenuWidget_PDA_EmailInbox.cpp
3>  MenuWidget_PDA_Objective.cpp
3>  MenuWidget_PDA_UserData.cpp
3>  MenuWidget_PDA_VideoInfo.cpp
3>  MenuWidget_Scrollbar.cpp
3>  MenuWidget_Shell_SaveInfo.cpp
3>  Clip.cpp
3>  Force.cpp
3>  Force_Constant.cpp
3>  Force_Drag.cpp
3>  Force_Field.cpp
3>  Force_Grab.cpp
3>  Force_Spring.cpp
3>  Physics.cpp
3>  Physics_Actor.cpp
3>  Physics_AF.cpp
3>  Physics_Base.cpp
3>  Physics_Monster.cpp
3>  Physics_Parametric.cpp
3>  Physics_Player.cpp
3>  Physics_RigidBody.cpp
3>  Physics_Static.cpp
3>  Physics_StaticMulti.cpp
3>  Push.cpp
3>  Script_Compiler.cpp
3>  Script_Interpreter.cpp
3>  Script_Program.cpp
3>  Script_Thread.cpp
3>  XA2_SoundHardware.cpp
3>  XA2_SoundSample.cpp
3>  XA2_SoundVoice.cpp
3>  i_sound_win32.cpp
3>  LightweightCompression.cpp
3>  PacketProcessor.cpp
3>  Snapshot.cpp
3>  SnapshotProcessor.cpp
3>  Snapshot_Jobs.cpp
3>  sys_achievements.cpp
3>  sys_dedicated_server_search.cpp
3>  sys_lobby.cpp
3>  sys_lobby_backend_direct.cpp
3>  sys_lobby_migrate.cpp
3>  sys_lobby_snapshot.cpp
3>  sys_lobby_users.cpp
3>  sys_local.cpp
3>  sys_localuser.cpp
3>  sys_profile.cpp
3>  sys_savegame.cpp
3>  sys_session_callbacks.cpp
3>  sys_session_local.cpp
3>  sys_session_savegames.cpp
3>  sys_signin.cpp
3>  sys_voicechat.cpp
3>  win_glimp.cpp
3>  win_input.cpp
3>  win_main.cpp
3>  win_shared.cpp
3>  win_snd.cpp
3>  win_syscon.cpp
3>  win_taskkeyhook.cpp
3>  win_wndproc.cpp
3>  win_cpu.cpp
4>------ Rebuild All started: Project: ALL_BUILD, Configuration: Release Win32 ------
4>  Building Custom Rule C:/RBDOOM-3-BFG-master/neo/CMakeLists.txt
4>  CMake does not need to re-run because C:\RBDOOM-3-BFG-master\build\CMakeFiles\generate.stamp is up-to-date.
5>------ Skipped Rebuild All: Project: INSTALL, Configuration: Release Win32 ------
5>Project not selected to build for this solution configuration 
========== Rebuild All: 3 succeeded, 1 failed, 1 skipped ==========

**Comment by kortemik** [comment]
Hello @deuxsonic 

We at Opentech Engine project have already done this:
https://github.com/OpenTechEngine/OpenTechBFG/pull/10/commits

-kordex

**Comment by deuxsonic** [comment]
Is it a fork of RBDOOM-3-BFG? What is different about it?

I Took a look at the GitHub website. Where do I obtain the Windows binaries? 
 


     On Sunday, October 18, 2015 3:17 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 Hello @deuxsonic We at Opentech Engine project have already done this:
https://github.com/OpenTechEngine/OpenTechBFG/pull/10/commits-kordex—

**Comment by kortemik** [comment]
Hello Ryan,

OpenTechEngine is a fork of RBDOOM-3-BFG. It removes doomclassics but is also a lot of other stuff as well. It's a standalone game engine project. On top of that it's work in progress.

Precompiled binaries for windows can be found at https://sourceforge.net/projects/opentechengine/files/snapshot/Windows/MinGW/

You need to copy the bfg base directory on top of shipped one.

Flashlight is the same as on rbdoom-3-bfg but it's not much work to convert it to the pistol one in case you'd like.

-Mikko

**Comment by deuxsonic** [comment]
So creating a game-agnostic engine. Does this change how it's parent game, DOOM 3 BFG, works with it?
 


     On Sunday, October 18, 2015 9:07 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 Hello Ryan,OpenTechEngine is a fork of RBDOOM-3-BFG. It removes doomclassics but is also a lot of other stuff as well. It's a standalone game engine project. On top of that it's work in progress.Precompiled binaries for windows can be found at https://sourceforge.net/projects/opentechengine/files/snapshot/Windows/MinGW/You need to copy the bfg base directory on top of shipped one.Flashlight is the same as on rbdoom-3-bfg but it's not much work to convert it to the pistol one in case you'd like.-Mikko—

**Comment by deuxsonic** [comment]
What do all the new assets for the base directory do? I don't recall other forks adding a bunch of assets as well.
 


     On Sunday, October 18, 2015 9:17 AM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 So creating a game-agnostic engine. Does this change how it's parent game, DOOM 3 BFG, works with it?
 


     On Sunday, October 18, 2015 9:07 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 Hello Ryan,OpenTechEngine is a fork of RBDOOM-3-BFG. It removes doomclassics but is also a lot of other stuff as well. It's a standalone game engine project. On top of that it's work in progress.Precompiled binaries for windows can be found at https://sourceforge.net/projects/opentechengine/files/snapshot/Windows/MinGW/You need to copy the bfg base directory on top of shipped one.Flashlight is the same as on rbdoom-3-bfg but it's not much work to convert it to the pistol one in case you'd like.-Mikko—

**Comment by deuxsonic** [comment]
Is there any way to put all the new assets in 1 file to keep the directory structure clean?
 


     On Sunday, October 18, 2015 9:21 AM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 What do all the new assets for the base directory do? I don't recall other forks adding a bunch of assets as well.
 


     On Sunday, October 18, 2015 9:17 AM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 So creating a game-agnostic engine. Does this change how it's parent game, DOOM 3 BFG, works with it?
 


     On Sunday, October 18, 2015 9:07 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 Hello Ryan,OpenTechEngine is a fork of RBDOOM-3-BFG. It removes doomclassics but is also a lot of other stuff as well. It's a standalone game engine project. On top of that it's work in progress.Precompiled binaries for windows can be found at https://sourceforge.net/projects/opentechengine/files/snapshot/Windows/MinGW/You need to copy the bfg base directory on top of shipped one.Flashlight is the same as on rbdoom-3-bfg but it's not much work to convert it to the pistol one in case you'd like.-Mikko—

**Comment by kortemik** [comment]
I haven't tried it but perhaps one could do on startup following:
 +set fs_resourceLoadPriority 0 +set fs_game path/to/bfg_base

**Comment by deuxsonic** [comment]
When I try to launch it gives me an error telling me it's not a valid Win32 executable. Is it 64-bit only?
 


     On Sunday, October 18, 2015 10:06 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 I haven't tried it but perhaps one could do on startup following:
 +set fs_resourceLoadPriority 0 +set fs_game path/to/bfg_base—

**Comment by deuxsonic** [comment]
Also I'm guessing it's a debug binary because of how huge it is. Can I get a 32-bit release binary please? I have VC++ 2010 Express and I wasn't able to build the RBDOOM-3-BFG stuff. I don't think I can even use the newer IDE.
 


     On Sunday, October 18, 2015 10:16 AM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 When I try to launch it gives me an error telling me it's not a valid Win32 executable. Is it 64-bit only?
 


     On Sunday, October 18, 2015 10:06 AM, Mikko Kortelainen <notifications@github.com> wrote:
   

 I haven't tried it but perhaps one could do on startup following:
 +set fs_resourceLoadPriority 0 +set fs_game path/to/bfg_base—

**Comment by Yetta1** [comment]
@deusxsonic To check if an executable is  32-bit or 64-bit, open it up in a text editor like notepad and look in around line 3 for something that will say something like "PE..d†", which the "d†" stands for 64-bit, if it shows an "L" it will be 32-bit. The RBDOOM3 and OpenTechEngine executables I have has "d†", so it's most likely that you have is 64-bit.

As Kortemik says, try using "+set fs_resourceLoadPriority 0 +set fs_game path/to/bfg_base" in the path line and set the directory where the resources has been copied or extracted to. There are console commands to extract the .resource files in the game, otherwise try Motorsep's mod launcher tool to extract the files.

http://www.kot-in-action.com/forum/viewtopic.php?f=28&t=4436&sid=7ef558e3c3e8d3d223e1fa341f7328dd

**Comment by deuxsonic** [comment]
Yes, using that method it's a 64-bit binary likely with all the debug symbols because it's like 80 MB. Basically I wanted to build RB-DOOM-3-BFG minus the doomclassic stuff but something's not right with my setup because I have the dependencies it needs and it's not compiling. Mikko steered me to OpenTechEngine which I assume is what I'm looking for (that being RB-DOOM-3-BFG without the doomclassic stuff) but I don't think I can build that either and I'd rather have a 32-bit and 64-bit release binaries, as debugging isn't something I feel I can do anyway. :(
 


     On Sunday, October 18, 2015 10:37 AM, Yetta1 <notifications@github.com> wrote:
   

 @deusxsonic To check if an executable is 32-bit or 64-bit, open it up in a text editor like notepad and look in around line 3 for something that will say something like "PE..d†", which the "d†" stands for 64-bit, if it shows an "L" it will be 32-bit. The RBDOOM3 and OpenTechEngine executables I have has "d†", so it's most likely that you have is 64-bit.As Kortemik says, try using "+set fs_resourceLoadPriority 0 +set fs_game path/to/bfg_base" in the path line and set the directory where the resources has been copied or extracted to. There are console commands to extract the .resource files in the game, otherwise try Motorsep's mod launcher tool to extract the files.http://www.kot-in-action.com/forum/viewtopic.php?f=28&t=4436&sid=7ef558e3c3e8d3d223e1fa341f7328dd—

**Comment by Yetta1** [comment]
@deuxsonic I'm not a programmer so I'm not sure how to remove the classic Doom stuff, but if you would like I could build the 32-bit binaries for you for RB and OTE on VS2013.

**Comment by deuxsonic** [comment]
That would be wonderful. Release 32-bit and 64-bit binaries of OTE are what I need. Thank you so much.
 


     On Sunday, October 18, 2015 11:07 AM, Yetta1 <notifications@github.com> wrote:
   

 @deuxsonic I'm not a programmer so I'm not sure how to remove the classic Doom stuff, but if you would like I could build the 32-bit binaries for you for RB and OTE on VS2013.—

**Comment by Yetta1** [comment]
@deuxsonic I've built the source but ran into some problems with the updated clones, for some reason the mouse movement does not work. Will upload the binaries once the problem has been solved.

**Comment by Yetta1** [comment]
Here are the builds for OTE, there is howver an issue with the engine crashing when the mousewheel is moved.
https://www.dropbox.com/s/zxfvz1nkqj0fgbk/OpenTechEngine.zip?dl=0

**Comment by deuxsonic** [comment]
Thank you again. Even the 32-bit build you provided tells me it's not a valid 32-bit application. They messed thing up good if it doesn't even run and will crash over something tiny like the mouse wheel. Could you possibly build the 32-bit RBDOOM-3-BFG release binary for me to see if it's also all messed up?
 


     On Sunday, October 18, 2015 2:01 PM, Yetta1 <notifications@github.com> wrote:
   

 Here are the builds for OTE, there is howver an issue with the engine crashing when the mousewheel is moved.
https://www.dropbox.com/s/zxfvz1nkqj0fgbk/OpenTechEngine.zip?dl=0—

**Comment by Yetta1** [comment]
The mouse wheel crash is related to weapon switching, currently OTE does not have any weapons added, that is mainly my fault as I'm working on the art assets for the engine and haven't added it into the project.

I am currently building a 32-bit binary of RB for you. I hope it will work, still not sure why the OTE binaries did not work for you.

**Comment by Yetta1** [comment]
Here is the Win32 binary for RB, there seems to be a problem with loading saved games or maps from the dev menu, starting a new campaign works.

https://www.dropbox.com/s/5n44drjwnbuyem1/RBDoom3BFG_Win32.zip?dl=0

**Comment by deuxsonic** [comment]
Was it compiled as release? I thought the dev menu only appeared in debug mode? Maybe I'm wrong.
 


     On Sunday, October 18, 2015 11:04 PM, Yetta1 <notifications@github.com> wrote:
   

 Here is the Win32 binary for RB, there seems to be a problem with loading saved games or maps from the dev menu, starting a new campaign works.https://www.dropbox.com/s/5n44drjwnbuyem1/RBDoom3BFG_Win32.zip?dl=0—

**Comment by deuxsonic** [comment]
Ha, turns out RB shows the same error. It's definitely my system.
 


     On Sunday, October 18, 2015 11:35 PM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 Was it compiled as release? I thought the dev menu only appeared in debug mode? Maybe I'm wrong.
 


     On Sunday, October 18, 2015 11:04 PM, Yetta1 <notifications@github.com> wrote:
   

 Here is the Win32 binary for RB, there seems to be a problem with loading saved games or maps from the dev menu, starting a new campaign works.https://www.dropbox.com/s/5n44drjwnbuyem1/RBDoom3BFG_Win32.zip?dl=0—

**Comment by deuxsonic** [comment]
Thanks for the help, now I know I have other problems on my hands. :)
 


     On Sunday, October 18, 2015 11:37 PM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 Ha, turns out RB shows the same error. It's definitely my system.
 


     On Sunday, October 18, 2015 11:35 PM, Ryan Paul Spidell <spamclark15@yahoo.com> wrote:
   

 Was it compiled as release? I thought the dev menu only appeared in debug mode? Maybe I'm wrong.
 


     On Sunday, October 18, 2015 11:04 PM, Yetta1 <notifications@github.com> wrote:
   

 Here is the Win32 binary for RB, there seems to be a problem with loading saved games or maps from the dev menu, starting a new campaign works.https://www.dropbox.com/s/5n44drjwnbuyem1/RBDoom3BFG_Win32.zip?dl=0—

**Comment by Yetta1** [comment]
It's built under release. I hardly ever build debug versions since i'm not sure what's going on on that side. I mainly work with 3D and graphics. The dev menu in RBDoom is a nice standard feature to quickly load maps.

Do you have DirectX SDK June 2010 installed? There seems to be an issue with some system that it  had to be installed before the Visual C++ 2010 redistributable packages, has there is a conflict between libraries. I've had that issue as well trying to build RB and OTE, I had to uninstall the Visual C++ X86 and X64 redistributables and the DirectX SDK june 2010, then reinstall the DX SDK followed by the VS C++ packages. There may also be a possibility that Cmake may not be in the system path.

**Comment by deuxsonic** [comment]
I have the DirectX SDK installed. Cmake is in the system path. The error it gives is actually a rather common error when I try to build. If you search for it, there are tons of cases of it popping up. I tried sticking the header it was looking for where VC++ would find it, and it got past that part but then produced more errors.
 


     On Sunday, October 18, 2015 11:50 PM, Yetta1 <notifications@github.com> wrote:
   

 It's built under release. I hardly ever build debug versions since i'm not sure what's going on on that side. I mainly work with 3D and graphics. The dev menu in RBDoom is a nice standard feature to quickly load maps.Do you have DirectX SDK June 2010 installed? There seems to be an issue with some system that it had to be installed before the Visual C++ 2010 redistributable packages, has there is a conflict between libraries. I've had that issue as well trying to build RB and OTE, I had to uninstall the Visual C++ X86 and X64 redistributables and the DirectX SDK june 2010, then reinstall the DX SDK followed by the VS C++ packages. There may also be a possibility that Cmake may not be in the system path.—

**Comment by Yetta1** [comment]
The only other solution I can think of is to download Visual Studio 2013 Community addition and try build it through that. Currently I have 2013 and 2015 running alongside each other, not sure if it would cause issues with 2010, otherwise try MinGW.

I still don't know why my builds do not work for you as I built it on a socket 775 Celeron E3300, it should work on newer systems. I've tested the builds on the Celeron system, on my Dual-core Pentium and my AMD system, both binaries worked fine.
